src/Command/App/CreateCdeCommand.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+namespace Acquia\Cli\Command\App;
+
+use Acquia\Cli\Command\CommandBase;
+use Acquia\Cli\Exception\AcquiaCliException;
+use Acquia\Cli\Helpers\LoopHelper;
+use Acquia\Cli\Output\Checklist;
+use AcquiaCloudApi\Connector\Client;
+use AcquiaCloudApi\Endpoints\Databases;
+use AcquiaCloudApi\Endpoints\Environments;
+use AcquiaCloudApi\Endpoints\Ides;
+use AcquiaCloudApi\Endpoints\Notifications;
+use AcquiaCloudApi\Response\EnvironmentResponse;
+use AcquiaCloudApi\Response\IdeResponse;
+use AcquiaCloudApi\Response\OperationResponse;
+use React\EventLoop\Loop;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Class CreateCdeCommand.
+ */
+class CreateCdeCommand extends CommandBase {
+
+  protected static $defaultName = 'app:environment:create';
+
+  /**
+   * @var \Acquia\Cli\Output\Checklist
+   */
+  private Checklist $checklist;
+
+  /**
+   * {inheritdoc}.
+   */
+  protected function configure() {
+    $this->setDescription('Create a new Continuous Delivery Environment (CDE)');
+    $this->addArgument('label', InputArgument::REQUIRED, 'The label of the new environment');
+    $this->addArgument('branch', InputArgument::OPTIONAL, 'The vcs path (git branch name) to deploy to the new environment');
+    $this->acceptApplicationUuid();
+  }
+
+  /**
+   * @param \Symfony\Component\Console\Input\InputInterface $input
+   * @param \Symfony\Component\Console\Output\OutputInterface $output
+   *
+   * @return int 0 if everything went fine, or an exit code
+   * @throws \Exception
+   */
+  protected function execute(InputInterface $input, OutputInterface $output) {
+    $this->output = $output;
+    $cloud_app_uuid = $this->determineCloudApplication(TRUE);
+    $label = $input->getArgument('label');
+    $acquia_cloud_client = $this->cloudApiClientService->getClient();
+    $environments_resource = new Environments($acquia_cloud_client);
+    $this->checklist = new Checklist($output);
+
+    $this->validateLabel($environments_resource, $cloud_app_uuid, $label);
+    $branch = $this->getBranch($acquia_cloud_client, $cloud_app_uuid, $input);
+    $database_names = $this->getDatabaseNames($acquia_cloud_client, $cloud_app_uuid);
+
+    $this->checklist->addItem("Initiating environment creation");
```

Comment:
We don't assert that any of these checklist items show up in output or get resolved